### PR TITLE
chore: hook for sensitive data leak

### PR DESCRIPTION
**Pull request by Jumas**
### Proposed changes

Restricts to committing the data containing urls to internal SBB resources (basically any text entries ending with ".sbb.ch" except for "www.sbb.ch" - which we usually use in the pom.xml).

### Checklist

Before creating a PR, run through this checklist and mark each as complete:
- [ ] I have read the [`CONTRIBUTING`](CONTRIBUTING.md) document
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,6 +16,13 @@ repos:
       - id: check-yaml
       - id: no-commit-to-branch
       - id: mixed-line-ending
+  - repo: local
+    hooks:
+      - id: sensitive-data-leak
+        name: Sensitive data leak hook
+        entry: (?<!www)\.sbb\.ch
```
Comment (ariwk) [inline .pre-commit-config.yaml:23]: it should check for:
- all sbb.ch except www.sbb.ch and polarion-opensource@sbb.ch
- u- and e-numbers

Comment (grigoriev) [reply .pre-commit-config.yaml]: Updated.
Must be split into 2 checks due to "look-behind requires fixed-width pattern" error.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -16,6 +16,19 @@ repos:
...
+    hooks:
+      - id: sensitive-data-leak-urls
+        name: Sensitive data leak - URLs
+        entry: sbb\.ch\b
+        language: pygrep
+        types: [text]
+        exclude: \b(www\.|polarion-opensource@)sbb\.ch\b
+      - id: sensitive-data-leak-ue-numbers
+        name: Sensitive data leak - UE numbers
+        entry: ([uU]|[eE])\d{6}
```
Comment (ariwk) [inline .pre-commit-config.yaml:29]: there are users like that too "ue10949"

Comment (grigoriev) [reply .pre-commit-config.yaml]: changed regexp for such case

**Review by ariwk — APPROVED** [review]

**Review by ariwk — APPROVED** [review]

**Pull request merged**